snap-confine: add new SC_CLEANUP and use it

mvo5 opened the pull request:
This makes reading the code (hopefully) slightly easier as it
shortens the __attribute__((cleanup(n))) use.

zyga commented on cmd/snap-confine/mount-support.c:
```diff
@@ -149,23 +149,14 @@ static void setup_private_pts()
 				    const char *snap_name)
 {
-	debug("%s: %s", __FUNCTION__, snap_name);
-
```
Some unexpected changes here, what is this about?

mvo5 commented:
Sorry, this was a bit of a mess (merge error and based on an old master from before the rally). I cleaned it up and squashed it, sorry again, this should be much cleaner now.
